web/src/Components/Optimization/PillInput.tsx

```diff
@@ -1,79 +1,104 @@
-import React, {ReactElement} from 'react';
-import styled from 'styled-components';
+import React, { ReactElement, useState } from 'react'
+import styled from 'styled-components'
 // @ts-ignore
-import {TextField} from '@equinor/eds-core-react';
+import { TextField, Button } from '@equinor/eds-core-react'
 
 const Wrapper = styled.div`
   display: grid;
   grid-template-rows: min-width;
   padding: 0px;
-  
   grid-gap: 2rem;
   position: relative;
   transition: all 0.36s;
   grid-template-columns: repeat(1, fit-content(100%));
 `
 
 export interface Pill {
-    volume: number
-    density: number
-    mass: number
+  volume: number
+  density: number
+  mass: number
 }
 
-
 interface PillInputProps {
-    isLoading: boolean
-    pill: Pill
-    setPill: Function
+  isLoading: boolean
+  pill: Pill
+  setPill: Function
+  handleOptimize: Function
 }
 
-const PillInput = ({pill, setPill, isLoading}: PillInputProps): ReactElement => {
+const PillInput = ({ pill, setPill, isLoading, handleOptimize }: PillInputProps): ReactElement => {
+  const [invalidVolume, setInvalidVolume] = useState<boolean>(false)
+  const [invalidDensity, setInvalidDensity] = useState<boolean>(false)
 
-    const handleVolumeChange = (event: any) => {
-        const volume = event.target.value
-        const mass = volume * pill.density
+  const handleChange = (type: string, value: string) => {
+    let newValue: number = 0
+    if (value !== '') newValue = parseInt(value)
 
-        setPill({
-            volume: volume,
-            mass: mass,
-            density: pill.density
-        })
+    if (Math.sign(newValue) <= 0) {
+      type === 'volume' && setInvalidVolume(true)
```

Comment:
If you set wrong (invalid) volume, then density is also shown wrong? However, density could be an ok value?

Reply:
Mass should be shown as invalid, but not sure that density should.

Reply:
Only the invalid property is shown as invalid.
Mass is not displayed.
User cant run optimization with invalid values.